src/server/db/mongo/user-privat-provider.js

```diff
@@ -0,0 +1,157 @@
+import UserPrivateModel from './models/user-private.js'
+import logger from '../../../imports/pino-logger'
+import { type UserRecord } from '../../../imports/types'
+
+class UserPrivate {
+  constructor() {
+    this.model = UserPrivateModel
+  }
+
+  /**
+   * Get new nonce after increment
+   *
+   * @param {UserRecord} user
+   *
+   * @returns {Promise<*>}
+   */
+  async isDupUserData(user: UserRecord): boolean {
+    const { email, mobile } = user
+    let result = null
+
+    if (email) {
+      result = await this.model.findOne({ email })
+      if (result) {
+        return true
+      }
+    }
+
+    if (mobile) {
+      result = await this.model.findOne({ mobile })
+      if (result) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  /**
+   * Create or update user privat date
+   *
+   * @param {UserRecord} user
+   *
+   * @returns {Promise<void>}
+   */
+  async updateUser(user: UserRecord): Promise<boolean> {
+    try {
+      const userDb = await this.getByIdentifier(user.identifier)
+      if (userDb) {
+        await this.model.update({ identifier: user.identifier }, { $set: user }, { new: true })
+      } else {
+        await this.model.create(user)
+      }
+      return true
+    } catch (ex) {
+      logger.error('Update user failed [mongo actions]:', { message: ex.message, user })
+    }
+
+    return false
+  }
+
+  /**
+   * Add new user
+   *
+   * @param {UserRecord} user
+   *
+   * @returns {Promise<boolean>}
+   */
+  async addUser(user: UserRecord): Promise<boolean> {
+    return this.updateUser(user)
+  }
+
+  /**
+   * Return row by field and value
+   *
+   * @param {string} identifier
+   * @param {string} field
+   *
+   * @returns {object || null}
+   */
+  async getUserField(identifier: string, field: string): string {
+    const result = await this.model.findOne({ identifier })
```

Comment:
add second parameter to get only the requested field

Reply:
done